### PR TITLE
bug(TDP-2925) Back - URL Encode the filename in full run result download

### PR DESCRIPTION
**Pull request by skermabon**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2925

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/299/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/299/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/300/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/300/)

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/ExportUtils.java [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
     }
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportUtils.class);
+
     public static void setExportHeaders(String exportName, ExportFormat format) {
         HttpResponseContext.contentType(format.getMimeType());
-        HttpResponseContext.header("Content-Disposition", "attachment; filename=\"" + exportName + format.getExtension() + "\"");
+        // TDP-2925 a multi-byte file name cannot export the file correctly
+        try {
+            HttpResponseContext.header("Content-Disposition",
```
Comment (hmdebenque) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/ExportUtils.java]: Please use the HttpHeaders.CONTENT_DISPOSITION constant

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/ExportUtils.java [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+                    "attachment; filename=\"" + UriUtils.encodePath(exportName, UTF_8.toString()) + format.getExtension() + "\"");
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.info("Can't encode '{}', will return the original name", exportName);
```
Comment (hmdebenque) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/ExportUtils.java]: UTF-8 is a charset that any JVM is required to support. If this happen we have a quite serious problem
See https://docs.oracle.com/javase/8/docs/api/java/nio/charset/Charset.html

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/302/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/302/)

**Pull request merged**